source/extensions/filters/network/zookeeper_proxy/decoder.cc

```diff
@@ -172,9 +172,11 @@ void DecoderImpl::decodeOnData(Buffer::Instance& data, uint64_t& offset) {
   }
 
   requests_by_xid_[xid] = {opcode, std::move(start_time)};
+
+  return opcode;
```

Comment:
I see that line 171 above throws an exception if the opcode is unknown. IIUC this method can be called on the data-path (not on the config). If so, please change the exception throwing to returning a nullopt or a value that signifies that an error occurred.

Reply:
This excaption will be caught later: https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/network/zookeeper_proxy/decoder.cc#L618-L632

Maybe we could keep it as is?

Reply:
Due to some security consideration, Envoy does not allow [exceptions on the data-path](https://github.com/envoyproxy/envoy/blob/a19966b92bc3afcdd68ebeffe53c4b6848733e79/STYLE.md?plain=1#L26). I suggest changing this behavior (in a different PR), to comply with the rest of Envoy

Reply:
Hi @adisuissa, here is the PR to remove EnvoyException from the ZK proxy filter data plane: https://github.com/envoyproxy/envoy/pull/30438

Could you elaborate why keeping EnvoyException in data plane will have the security issues? Thanks!

Reply:
There's prior work on removing exceptions from the data-plane, e.g., #10878.